src/GitHub.Api/Tasks/ActionTask.cs

```diff
@@ -5,6 +5,53 @@
 
 namespace GitHub.Unity
 {
+    class TaskQueue : TaskBase
+    {
+        private TaskCompletionSource<bool> aggregateTask = new TaskCompletionSource<bool>();
+        private readonly List<ITask> queuedTasks = new List<ITask>();
+        private volatile bool isSuccessful = true;
+        private volatile Exception exception;
+        private int finishedTaskCount;
+
+        public TaskQueue() : base()
+        {
+            Initialize(aggregateTask.Task);
+        }
+
+        public ITask Queue(ITask task)
+        {
+            task.OnEnd += TaskFinished;
+            queuedTasks.Add(task);
+            return this;
+        }
+
+        protected override void Run()
+        {
+            foreach (var task in queuedTasks)
+                task.Start();
+            base.Run();
+        }
+
+        private void TaskFinished(ITask task, bool success, Exception ex)
+        {
+            var count = Interlocked.Increment(ref finishedTaskCount);
+            isSuccessful &= success;
+            if (!success)
+                exception = ex;
+            if (count == queuedTasks.Count)
+            {
+                if (isSuccessful)
+                {
+                    aggregateTask.TrySetResult(true);
+                }
+                else
+                {
+                    aggregateTask.TrySetException(ex);
```

Comment:
this should be exception rather than ex

Reply:
Super awesome that you're so responsive. Two out of three bugs closed already, however when is the next release? I'd like to present this to our teams on Friday, but I'd like the two closed bugs in before I do. About the race condition here I'm not super worried about it... the task completes and the icons will update correctly, that the dialog error message fails once in a thousand is hardly a blocker for me as long as TrySetException(null) doesn't blow up in our face. 
:thumbsup: